src/Repository/CardRepository.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace App\Repository;
+
+use App\Entity\Card;
+use Doctrine\ORM\EntityManagerInterface;
+use App\Request\CreateCard;
+
+class CardRepository
+{
+    private EntityManagerInterface $entityManager;
+
+    public function __construct(EntityManagerInterface $entityManager)
+    {
+        $this->entityManager = $entityManager;
+    }
+
+    public function add(Card $card): void
+    {
+        $this->entityManager->persist($card);
+    }
+
+    public function findOneByName(string $name): ?Card
+    {
+        return $this->entityManager->getRepository(Card::class)->findOneBy(['name' => $name]);
+    }
+
+    public function createCard(CreateCard $createCard): void
```

Comment:
This is not correct place to convert DTO into Entity. In basic you put responsibility of creating Entity object in repository but this responsibility is not related to this class.

Thing about it. How many reasons for change we currently have in repository? 
1. One is when something change related to fetching entities from DB.
2. Next one is when we change something what is related to Create Entity for example we add new property.

So if we have more then one reason to change there is a chance that we do something wrong. Here IMO we brake SRP of repository. In some cases we don't follow that rule but we need to good reason for it.

We have multiple ways to make it better.:
1. Put it into controller. Why? Cuz this logic is simple and if we don't better solution then make it simply and strait forward. And from there you could look on it and thing once more how to refactor it.
2. Create handler like`CreateCardHandler::handle(CreateCard): void`, that handle Domain Validation, Converting DTO to Entity and invoke `add` method on repository.

I suggest option one for now. If we gather more examples then we make a call and discuss it.

Reply:
done